packages/client/src/prax.ts

```diff
@@ -22,7 +22,10 @@ export class PraxManifestError extends Error {}
 
 export const getPraxPort = async () => {
   const provider = window[PenumbraSymbol]?.[prax_origin];
-  if (!provider) throw new Error('Prax not installed');
+  if (!provider) throw new PraxNotInstalledError('Prax not installed');
+
+  await throwIfPraxNotConnectedTimeout();
+
   return provider.connect();
```

Comment:
Okay, if we remove `throwIfPraxNotConnectedTimeout()` from all loders and call it inside `getPraxPort()`, that solves all issues 

The only thing I don't really like is that we basically check that prax is connected first, and only then do we call `provider.connect()`
How can we be connected if we never call `connect()`?

Reply:
connection state is available from `isConnected` on the global. connection may resolve and activate before the actual `connect` function is called, if the page is already approved.